webapp/components/survey/NodeDefDetails/store/actions/useValidateNodeDef.js

```diff
@@ -0,0 +1,35 @@
+import * as R from 'ramda'
+
+import * as Survey from '@core/survey/survey'
+import * as SurveyValidator from '@core/survey/surveyValidator'
+
+import * as SurveyState from '@webapp/store/survey/state'
+
+import * as NodeDefState from '../state'
+
+import types from './types'
+
+export const useValidateNodeDef = ({ nodeDefState, setNodeDefState }) => ({ nodeDef }) => async (
+  dispatch,
+  getState
+) => {
+  const state = getState()
+  const survey = SurveyState.getSurvey(state)
+
+  // Update survey with updated node def
+  const surveyUpdated = R.pipe(
+    // Associate updated node def
+    Survey.assocNodeDef(nodeDef),
+    // Build and associate dependency graph
+    Survey.buildAndAssocDependencyGraph
+  )(survey)
```

Comment:
how about to extract this to local function like:

```
const assocNodeDefAndDependencyGraph = ({ nodeDef }) => (survey) => ...
```

in that way, the code could be more readable

Reply:
uhm, you are right, I'll extract that part in a function

Reply:
or even better pass a parameter:

```const assocNodeDef = ({ nodeDef, updateDependencyGraph: false }) => (survey) => ...```

Reply:
done